holidayextras/countdown-notify-lambda
# Use index

#### What does this PR do? (please provide any background)

Use a `query` operation using the new secondary index on the database, this speeds up finding notifications sooo much!

#### What tests does this PR have?

Current unit tests updated.

#### How can this be tested?

1. Setup the development AWS environment: `AWS_ACCESS_KEY_ID=x AWS_SECRET_ACCESS_KEY=x npm AWS_ACCOUNT=x run build`
1. Edit the dynamo db to have an event in the next hour.
1. Run the lambda function: `npm start`

You will see the event being found and then not being sent (which is expected).

#### Any tech debt?

Using the schema version as a partition key is not optimal but does allow us to do a range check on a sort key.

#### Screenshots / Screencast

#### What gif best describes how you feel about this work?

![giphy](https://cloud.githubusercontent.com/assets/3158640/21771499/7b6fdc54-d67f-11e6-89fa-dd2607612801.gif)

- I have checked our general [contributing document](https://github.com/holidayextras/culture/blob/master/CONTRIBUTING.md) and the project specific [contributing document](../blob/master/CONTRIBUTING.md) (if present) and I'm happy for this to be reviewed.

By approving a review you are confirming you have...
- Witnessed the work behaving as expected (this could be on the author's machine or screencast).
- Checked for coding anti-patterns.
- Checked for appropriate test coverage.
- Checked all the tests are passing.

---
